### DISCUSSION TITLE
Workaround for binary format size limitations?

### DISCUSSION THREAD

**Discussion by jcnewell** [explosion/spaCy#7061] [Help: Coding & Implementations] [answered]
In switching to the binary training format, I'm coming up against a known problem, which is the max size limit of the serialiser, leading to an error when calling DocBin.to_disk

    line 206, in srsly.msgpack._packer.Packer._pack ValueError: bytes object is too large

Is there a workaround for "spacy train"  e.g. loading multiple binary training files or reverting to the JSON file format?

**Comment by adrianeboyd** [comment, marked as answer]
If you provide a directory as the `path` (same as `paths.train` or `paths.dev`), the `Corpus` reader searches recursively for `.spacy` files in that directory, so you should be able to split your dataset into multiple `.spacy` files.

If this isn't quite what you want to do, it shouldn't be too difficult to write a custom reader, either. Here's what `Corpus` looks like:

https://github.com/explosion/spaCy/blob/1e3a326e5362eee73b0b7dacb89994a2507390b7/spacy/training/corpus.py#L82-L193

**Reply by jcnewell** [reply]
Thanks Adriane - this solution is working well.
